geogem.rb

```diff
@@ -45,9 +45,20 @@ def self_intersects?
   end
 
   def intersects?(poly)
-    return true unless !bbox_intersect?(self.bbox, poly.bbox)
-    return true unless !poly.nodes.each { |n| self.point_in_poly?(n) }.any?
-    self.nodes.each { |n| poly.point_in_poly?(n) }.any?
+    #binding.pry
+    return false unless bbox_intersect?(self.bbox, poly.bbox)
+    return true unless !poly.nodes.map { |n| point_in_poly?(n) }.any?
+    nodes.map { |n| poly.point_in_poly?(n) }.any?
+  end
+
+  protected 
+
+  def point_in_poly?(test_point)
```

Comment:
Potentially make your variable names a little more descriptive. For things like bbox1 it's fine because you use them a lot throughout the code, but intersection_cnt means nothing to the user on it's own (I can see it's a count on line 60 but, yaknow)

Reply:
Okies, I'll try to stop shrinking my variable :)